### PR TITLE
NuGet package

### LINKED ISSUE #297: Please make a NuGet package. That can help add your library to project easier

### PR DESCRIPTION
**Pull request by simon-berger**
# Work in progress!
Do not merge.

# Issue
Close #297 

# TODOs
- [x] Locally create working package
- [x] Automate package generation (CI)
- [x] (Automated) publishing of the created package
- [x] Fix uploaded version of 1.9.0
- [x] Update Readme

# Description
Because of the poor documentation from Microsoft, I mainly used this manual as a guide: [link](https://digitalhouseblog.wordpress.com/2019/08/22/how-to-make-a-nuget-package-for-c/)
With this, I managed to create a native NuGet package which automatically gets created from a CI job if a new release tag is created. The created package is then uploaded as a job artifact and published to NuGet.org.
Here is an example of the CI action with an working package build: [example CI run](https://github.com/simon-berger/cpr/actions/runs/2699193424)

### CODE REVIEW

**Comment by simon-berger** [comment]
I have successfully created a CI job which automatically creates the NuGet package and uploads the result as an artifact ([example run](https://github.com/simon-berger/cpr/actions/runs/2699193424)) 🎉 .
The next step would be to (automatically) publish the package (if intended) to NuGet.org, for which I would need a NuGet account and API key.
Moreover we maybe need to change certain metadata in the .nuspec file (e.g. Who should be the author?). 

@COM8 can you please please briefly check the metadata and let me know if you want the package to be published.

File: .github/workflows/build-nuget.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+name: Build NuGet Package
+on:
```
Comment (COM8) [inline .github/workflows/build-nuget.yml:2]: These triggers are nice!
Could you please also add the `workflow_dispatch` trigger, so one could rebuild the package by manually triggering it.

References:
* https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/
* https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

File: nuget/libcpr.nuspec [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>libcpr</id>
+    <version>$VERSION$</version>
+    <title>C++ Requests: Curl for People</title>
+    <authors>Simon Berger</authors>
+    <owners></owners>
```
Comment (COM8) [inline nuget/libcpr.nuspec:8]: Please add "Fabian Sauter", "Kilian Traub" and "many other libcpr contributors" as owners

**Review by COM8 — CHANGES_REQUESTED** [review]
Looks overall good.

**Comment by COM8** [comment]
In my eyes it makes sense the you use an API-Key I provide as "libcpr" to make sure we can even in the future change stuff if required directly.

**Comment by COM8** [comment]
I have created an account for `nuget@libcpr.org`. Like for the .deb package I can forward all mail arriving there to you in case you are interested.

**Review by COM8 — APPROVED** [review]

**Comment by simon-berger** [comment]
I completed testing with [this](https://github.com/simon-berger/cpr/suites/7522670883/artifacts/309779147) fresh package.

**Review by COM8 — APPROVED** [review]
Great work! Thanks.

**Comment by COM8** [comment]
The CI worked perfectly: https://www.nuget.org/packages/libcpr/#versions-body-tab

**Comment by simon-berger** [comment]
Just got the email, perfect 🎉

**Pull request merged**